irods_specific_implementation/irods_rest_logical_path_api_implementation.h

```diff
@@ -115,10 +119,371 @@ namespace irods::rest
+                error("Caught exception - [error_code={}] {}", e.code(), e.what());
+                return make_error_response(e.code(), e.what());
+            }
+            catch (const std::exception& e) {
+                error("Caught exception - {}", e.what());
+                return make_error_response(SYS_INVALID_INPUT_PARAM, e.what());
+            }
+	    catch (...){
+		error("Caught unknown exception.");
+		return make_error_response(SYS_UNKNOWN_ERROR, "Caught unknown exception");
+	    }
+        }
+
+        std::tuple<Pistache::Http::Code, std::string>
+        replicate_dispatcher(const Pistache::Rest::Request& _request,
+                   Pistache::Http::ResponseWriter& _response)
+        {
+            try {
+              dataObjInp_t repl_input{};
+
+              auto conn = get_connection(_request.headers().getRaw("authorization").value());
+
+              bool is_collection = false;
+
+              setup_input_for_repl(
+                  repl_input,
+                  _request,
+                  *conn(),
+                  is_collection
+              );
+
+              int status;
+              const auto start_path = repl_input.objPath;
+              if ( is_collection ) {
+                if ( ! is_set(_request.query().get("recursive").getOrElse("")) )
+                {
+                    return std::make_tuple(
+                        Pistache::Http::Code::Bad_Request,
+                        "'recursive=1' required to replicate a collection. Make sure you want to replicate the whole sub-tree."
+                    );
+                }
+
+                for ( const auto& pathname : fscli::recursive_collection_iterator(*conn(), start_path) ) {
+                    if ( fscli::is_data_object(fscli::status(*conn(), decode_url(pathname.path()))) ) {
+			strcpy(
+			  repl_input.objPath,
+			  pathname.path().c_str()
+			);
+
+                        status = rcDataObjRepl(conn(), &repl_input);
+                        if ( status == SYS_INVALID_FILE_PATH ) {
+                            return make_error_response(status, "You passed an invalid path for replication. Make sure that you are giving an absolute path.");
+                        }
+                        else if ( status == SYS_NOT_ALLOWED ) {
+                            std::stringstream msg;
+                            msg << "You have tried to repl an object in a way that is not allowed.\n";
+                            msg << "Some possible causes are:\n";
+                            msg << "- You may have tried to replicate a non-good replica.\n";
+                            msg << "- The object you tried to replicate may have a checksum that wasn't verified.\n";
+                            msg << "- You may have tried to replicate a data object onto an already identical replica.\n";
+                            msg << "Please check for these issues and contact your administrator if the issue isn't resolved.\n";
```

Comment:
If I'm recalling correctly, survey says that since this is a developer interface anyway, end-users shouldn't even get messages like these, so all these "user-facing" messages will just go away. @trel ?

Reply:
guessing never seems... quite right.   better to just send the real deal... usually.

however, the user in this case is the app developer (or curl itself) - so messages aren't bad... just have to always balance verbosity/helpfulness vs security/bandwidth/pollution.

let's leave them here for now, we're still pre-1.0.